Dependencies upgrade and security vulnerabilities fix in bis-schemas, bis-schema-validation and ecjson2md repositories

There are several critical and high level security vulnerabilities reported:

for example audit currently fails on:

```
Running audit
Audit time: 6.323s

##[error]HIGH Security Vulnerability: @isaacs/brace-expansion has Uncontrolled Resource Consumption in @isaacs/brace-expansion (from .>@itwin/build-tools>@microsoft/api-extractor>minimatch>@isaacs/brace-expansion).  See https://github.com/advisories/GHSA-7h2j-956f-4vf2 for more info.
##[error]HIGH Security Vulnerability: fast-xml-parser has RangeError DoS Numeric Entities Bug in fast-xml-parser (from .>@itwin/core-backend>@azure/storage-blob>@azure/core-xml>fast-xml-parser).  See https://github.com/advisories/GHSA-37qj-frw5-hhjh for more info.
##[error]CRITICAL Security Vulnerability: fast-xml-parser has an entity encoding bypass via regex injection in DOCTYPE entity names in fast-xml-parser (from .>@itwin/core-backend>@azure/storage-blob>@azure/core-xml>fast-xml-parser).  See https://github.com/advisories/GHSA-m7jm-9gc2-mpf2 for more info.
##[error]HIGH Security Vulnerability: fast-xml-parser affected by DoS through entity expansion in DOCTYPE (no expansion limit) in fast-xml-parser (from .>@itwin/core-backend>@azure/storage-blob>@azure/core-xml>fast-xml-parser).  See https://github.com/advisories/GHSA-jmr7-xgp7-cmfj for more info.
##[error]HIGH Security Vulnerability: fast-xml-parser affected by numeric entity expansion bypassing all entity expansion limits (incomplete fix for CVE-2026-26278) in fast-xml-parser (from .>@itwin/core-backend>@azure/storage-blob>@azure/core-xml>fast-xml-parser).  See https://github.com/advisories/GHSA-8gc5-j5rx-235r for more info.
##[warning]MODERATE Security Vulnerability: markdown-it is has a Regular Expression Denial of Service (ReDoS) in markdown-it (from .>@itwin/build-tools>typedoc>markdown-it).  See https://github.com/advisories/GHSA-38c4-r59v-3vqw for more info.
##[error]HIGH Security Vulnerability: Axios is Vulnerable to Denial of Service via __proto__ Key in mergeConfig in axios (from .>@itwin/core-backend>@itwin/object-storage-azure>@itwin/object-storage-core>axios).  See https://github.com/advisories/GHSA-43fc-jf86-j433 for more info.
##[error]HIGH Security Vulnerability: minimatch has a ReDoS via repeated wildcards with non-matching literal in pattern in minimatch (from .>@types/rimraf>@types/glob>glob>minimatch).  See https://github.com/advisories/GHSA-3ppc-4f35-3m26 for more info.
##[error]HIGH Security Vulnerability: minimatch has ReDoS: matchOne() combinatorial backtracking via multiple non-adjacent GLOBSTAR segments in minimatch (from .>@types/rimraf>@types/glob>glob>minimatch).  See https://github.com/advisories/GHSA-7r86-cg39-jmmj for more info.
##[error]HIGH Security Vulnerability: minimatch ReDoS: nested *() extglobs generate catastrophically backtracking regular expressions in minimatch (from .>@types/rimraf>@types/glob>glob>minimatch).  See https://github.com/advisories/GHSA-23c5-xmqv-rm74 for more info.
##[error]CRITICAL Security Vulnerability: fast-xml-parser has an entity encoding bypass via regex injection in DOCTYPE entity names in fast-xml-parser (from .>@itwin/imodels-access-backend>@itwin/object-storage-google>@google-cloud/storage>fast-xml-parser).  See https://github.com/advisories/GHSA-m7jm-9gc2-mpf2 for more info.
##[error]HIGH Security Vulnerability: fast-xml-parser affected by DoS through entity expansion in DOCTYPE (no expansion limit) in fast-xml-parser (from .>@itwin/imodels-access-backend>@itwin/object-storage-google>@google-cloud/storage>fast-xml-parser).  See https://github.com/advisories/GHSA-jmr7-xgp7-cmfj for more info.
##[error]HIGH Security Vulnerability: fast-xml-parser affected by numeric entity expansion bypassing all entity expansion limits (incomplete fix for CVE-2026-26278) in fast-xml-parser (from .>@itwin/imodels-access-backend>@itwin/object-storage-google>@google-cloud/storage>fast-xml-parser).  See https://github.com/advisories/GHSA-8gc5-j5rx-235r for more info.
##[error]HIGH Security Vulnerability: flatted vulnerable to unbounded recursion DoS in parse() revive phase in flatted (from .>eslint>file-entry-cache>flat-cache>flatted).  See https://github.com/advisories/GHSA-25h7-pfq9-p65f for more info.
##[error]HIGH Security Vulnerability: minimatch has a ReDoS via repeated wildcards with non-matching literal in pattern in minimatch (from .>@itwin/eslint-plugin>eslint-plugin-import>minimatch).  See https://github.com/advisories/GHSA-3ppc-4f35-3m26 for more info.
##[error]HIGH Security Vulnerability: minimatch has a ReDoS via repeated wildcards with non-matching literal in pattern in minimatch (from .>@itwin/build-tools>glob>minimatch).  See https://github.com/advisories/GHSA-3ppc-4f35-3m26 for more info.
##[error]HIGH Security Vulnerability: minimatch has a ReDoS via repeated wildcards with non-matching literal in pattern in minimatch (from .>@itwin/eslint-plugin>@typescript-eslint/eslint-plugin>@typescript-eslint/type-utils>@typescript-eslint/typescript-estree>minimatch).  See https://github.com/advisories/GHSA-3ppc-4f35-3m26 for more info.
##[error]HIGH Security Vulnerability: minimatch has a ReDoS via repeated wildcards with non-matching literal in pattern in minimatch (from .>@itwin/build-tools>@microsoft/api-extractor>minimatch).  See https://github.com/advisories/GHSA-3ppc-4f35-3m26 for more info.
##[error]HIGH Security Vulnerability: minimatch has ReDoS: matchOne() combinatorial backtracking via multiple non-adjacent GLOBSTAR segments in minimatch (from .>@itwin/eslint-plugin>eslint-plugin-import>minimatch).  See https://github.com/advisories/GHSA-7r86-cg39-jmmj for more info.
##[error]HIGH Security Vulnerability: minimatch has ReDoS: matchOne() combinatorial backtracking via multiple non-adjacent GLOBSTAR segments in minimatch (from .>@itwin/build-tools>glob>minimatch).  See https://github.com/advisories/GHSA-7r86-cg39-jmmj for more info.
##[error]HIGH Security Vulnerability: minimatch has ReDoS: matchOne() combinatorial backtracking via multiple non-adjacent GLOBSTAR segments in minimatch (from .>@itwin/eslint-plugin>@typescript-eslint/eslint-plugin>@typescript-eslint/type-utils>@typescript-eslint/typescript-estree>minimatch).  See https://github.com/advisories/GHSA-7r86-cg39-jmmj for more info.
##[error]HIGH Security Vulnerability: minimatch has ReDoS: matchOne() combinatorial backtracking via multiple non-adjacent GLOBSTAR segments in minimatch (from .>@itwin/build-tools>@microsoft/api-extractor>minimatch).  See https://github.com/advisories/GHSA-7r86-cg39-jmmj for more info.
##[error]HIGH Security Vulnerability: minimatch ReDoS: nested *() extglobs generate catastrophically backtracking regular expressions in minimatch (from .>@itwin/eslint-plugin>eslint-plugin-import>minimatch).  See https://github.com/advisories/GHSA-23c5-xmqv-rm74 for more info.
##[error]HIGH Security Vulnerability: minimatch ReDoS: nested *() extglobs generate catastrophically backtracking regular expressions in minimatch (from .>@itwin/build-tools>glob>minimatch).  See https://github.com/advisories/GHSA-23c5-xmqv-rm74 for more info.
##[error]HIGH Security Vulnerability: minimatch ReDoS: nested *() extglobs generate catastrophically backtracking regular expressions in minimatch (from .>@itwin/eslint-plugin>@typescript-eslint/eslint-plugin>@typescript-eslint/type-utils>@typescript-eslint/typescript-estree>minimatch).  See https://github.com/advisories/GHSA-23c5-xmqv-rm74 for more info.
##[error]HIGH Security Vulnerability: minimatch ReDoS: nested *() extglobs generate catastrophically backtracking regular expressions in minimatch (from .>@itwin/build-tools>@microsoft/api-extractor>minimatch).  See https://github.com/advisories/GHSA-23c5-xmqv-rm74 for more info.
##[error]HIGH Security Vulnerability: Serialize JavaScript is Vulnerable to RCE via RegExp.flags and Date.prototype.toISOString() in serialize-javascript (from .>mocha>serialize-javascript).  See https://github.com/advisories/GHSA-5c6j-r48x-rmvq for more info.
##[warning]MODERATE Security Vulnerability: ajv has ReDoS when using `$data` option in ajv (from .>eslint>ajv).  See https://github.com/advisories/GHSA-2g4f-4pwh-qvx6 for more info.
##[warning]MODERATE Security Vulnerability: ajv has ReDoS when using `$data` option in ajv (from .>@itwin/build-tools>@microsoft/api-extractor>@microsoft/tsdoc-config>ajv).  See https://github.com/advisories/GHSA-2g4f-4pwh-qvx6 for more info.
##[warning]MODERATE Security Vulnerability: ajv has ReDoS when using `$data` option in ajv (from .>@itwin/build-tools>@microsoft/api-extractor>@rushstack/node-core-library>ajv).  See https://github.com/advisories/GHSA-2g4f-4pwh-qvx6 for more info.
##[error]HIGH Security Vulnerability: Underscore has unlimited recursion in _.flatten and _.isEqual, potential for DoS attack in underscore (from .>httpntlm>underscore).  See https://github.com/advisories/GHSA-qpx9-hpmf-5gmw for more info.
Finishing: Rush audit
```

--- a/README.md
+++ b/README.md
@@ -13,7 +13,7 @@ Each package has its own **node_modules** directory that contains symbolic links
 ## Prerequisites
 
 - [Git](https://git-scm.com/)
-- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 22. The Node installation also includes the **npm** package manager.
+- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 24. The Node installation also includes the **npm** package manager.
 - [Rush](https://github.com/Microsoft/web-build-tools/wiki/Rush): to install `npm install -g @microsoft/rush`
 - [TypeScript](https://www.typescriptlang.org/): this is listed as a devDependency, so if you're building it from source, you will get it with `rush install`.
 - [Visual Studio Code](https://code.visualstudio.com/): an optional dependency, but the repository structure is optimized for its use

--- a/README.md
+++ b/README.md
@@ -13,7 +13,7 @@ Each package has its own **node_modules** directory that contains symbolic links
 ## Prerequisites
 
 - [Git](https://git-scm.com/)
-- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 24. The Node installation also includes the **npm** package manager.
+- [Node](https://nodejs.org/en/): an installation of Node 24.15. The Node installation also includes the **npm** package manager.
 - [Rush](https://github.com/Microsoft/web-build-tools/wiki/Rush): to install `npm install -g @microsoft/rush`
 - [TypeScript](https://www.typescriptlang.org/): this is listed as a devDependency, so if you're building it from source, you will get it with `rush install`.
 - [Visual Studio Code](https://code.visualstudio.com/): an optional dependency, but the repository structure is optimized for its use

--- a/README.md
+++ b/README.md
@@ -13,7 +13,7 @@ Each package has its own **node_modules** directory that contains symbolic links
 ## Prerequisites
 
 - [Git](https://git-scm.com/)
-- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 14. The Node installation also includes the **npm** package manager.
+- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 22. The Node installation also includes the **npm** package manager.
 - [Rush](https://github.com/Microsoft/web-build-tools/wiki/Rush): to install `npm install -g @microsoft/rush`
 - [TypeScript](https://www.typescriptlang.org/): this is listed as a devDependency, so if you're building it from source, you will get it with `rush install`.
 - [Visual Studio Code](https://code.visualstudio.com/): an optional dependency, but the repository structure is optimized for its use

--- a/app-schema-validator/README.md
+++ b/app-schema-validator/README.md
@@ -8,7 +8,7 @@ The purpose of this NPM CLI tool is to validate the schemas present within an ap
 
 ### Prerequisites
 
-- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 20. The Node installation also includes the **npm** package manager.
+- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 22. The Node installation also includes the **npm** package manager.
 
 ### Installation
 

--- a/app-schema-validator/README.md
+++ b/app-schema-validator/README.md
@@ -8,7 +8,7 @@ The purpose of this NPM CLI tool is to validate the schemas present within an ap
 
 ### Prerequisites
 
-- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 24. The Node installation also includes the **npm** package manager.
+- [Node](https://nodejs.org/en/): an installation of Node 24.15. The Node installation also includes the **npm** package manager.
 
 ### Installation
 

--- a/app-schema-validator/README.md
+++ b/app-schema-validator/README.md
@@ -8,7 +8,7 @@ The purpose of this NPM CLI tool is to validate the schemas present within an ap
 
 ### Prerequisites
 
-- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 22. The Node installation also includes the **npm** package manager.
+- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 24. The Node installation also includes the **npm** package manager.
 
 ### Installation
 

--- a/common/config/azure-pipelines/buildPublishPackages.yaml
+++ b/common/config/azure-pipelines/buildPublishPackages.yaml
@@ -25,9 +25,9 @@ stages:
             clean: all
           - checkout: bis-schemas
           - task: NodeTool@0
-            displayName: "Install Node 22.x"
+            displayName: "Install Node 24.x"
             inputs:
-              versionSpec: 22.x
+              versionSpec: 24.x
               checkLatest: true
 
           - script: node $(System.DefaultWorkingDirectory)/bis-schema-validation/common/scripts/install-run-rush.js check

--- a/common/config/azure-pipelines/buildPublishPackages.yaml
+++ b/common/config/azure-pipelines/buildPublishPackages.yaml
@@ -25,9 +25,9 @@ stages:
             clean: all
           - checkout: bis-schemas
           - task: NodeTool@0
-            displayName: "Install Node 24.x"
+            displayName: "Install Node 24.15.0"
             inputs:
-              versionSpec: 24.x
+              versionSpec: 24.15.0
               checkLatest: true
 
           - script: node $(System.DefaultWorkingDirectory)/bis-schema-validation/common/scripts/install-run-rush.js check

--- a/common/config/azure-pipelines/version-bump.yaml
+++ b/common/config/azure-pipelines/version-bump.yaml
@@ -16,9 +16,9 @@ steps:
   - script: 'git checkout --track origin/master'
     displayName: 'Switching to master '
   - task: NodeTool@0
-    displayName: 'Install Node 24.x'
+    displayName: 'Install Node 24.15.0'
     inputs:
-      versionSpec: 24.x
+      versionSpec: 24.15.0
       checkLatest: true
 
   - script: node $(System.DefaultWorkingDirectory)/common/scripts/install-run-rush.js version --bump

--- a/common/config/azure-pipelines/version-bump.yaml
+++ b/common/config/azure-pipelines/version-bump.yaml
@@ -16,9 +16,9 @@ steps:
   - script: 'git checkout --track origin/master'
     displayName: 'Switching to master '
   - task: NodeTool@0
-    displayName: 'Install Node 22.x'
+    displayName: 'Install Node 24.x'
     inputs:
-      versionSpec: 22.x
+      versionSpec: 24.x
       checkLatest: true
 
   - script: node $(System.DefaultWorkingDirectory)/common/scripts/install-run-rush.js version --bump

--- a/imodel-schema-validator/README.md
+++ b/imodel-schema-validator/README.md
@@ -9,7 +9,7 @@ The purpose of this NPM CLI tool is to validate the schemas present within an iM
 ### Prerequisites
 
 - [Git](https://git-scm.com/)
-- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 14. The Node installation also includes the **npm** package manager.
+- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 22. The Node installation also includes the **npm** package manager.
 - [TypeScript](https://www.typescriptlang.org/): this is listed as a devDependency, so if you're building it from source, you will get it with `rush install`.
 - [Visual Studio Code](https://code.visualstudio.com/): an optional dependency, but the repository structure is optimized for its use
 

--- a/imodel-schema-validator/README.md
+++ b/imodel-schema-validator/README.md
@@ -9,7 +9,7 @@ The purpose of this NPM CLI tool is to validate the schemas present within an iM
 ### Prerequisites
 
 - [Git](https://git-scm.com/)
-- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 24. The Node installation also includes the **npm** package manager.
+- [Node](https://nodejs.org/en/): an installation of Node 24.15. The Node installation also includes the **npm** package manager.
 - [TypeScript](https://www.typescriptlang.org/): this is listed as a devDependency, so if you're building it from source, you will get it with `rush install`.
 - [Visual Studio Code](https://code.visualstudio.com/): an optional dependency, but the repository structure is optimized for its use
 

--- a/imodel-schema-validator/README.md
+++ b/imodel-schema-validator/README.md
@@ -9,7 +9,7 @@ The purpose of this NPM CLI tool is to validate the schemas present within an iM
 ### Prerequisites
 
 - [Git](https://git-scm.com/)
-- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 22. The Node installation also includes the **npm** package manager.
+- [Node](https://nodejs.org/en/): an installation of the latest security patch of Node 24. The Node installation also includes the **npm** package manager.
 - [TypeScript](https://www.typescriptlang.org/): this is listed as a devDependency, so if you're building it from source, you will get it with `rush install`.
 - [Visual Studio Code](https://code.visualstudio.com/): an optional dependency, but the repository structure is optimized for its use
 

--- a/rush.json
+++ b/rush.json
@@ -2,7 +2,7 @@
   "$schema": "https://developer.microsoft.com/json-schemas/rush/v5/rush.schema.json",
   "rushVersion": "5.147.0",
   "pnpmVersion": "9.15.0",
-  "nodeSupportedVersionRange": "^20.11.0 || ^22.11.0",
+  "nodeSupportedVersionRange": "^20.11.0 || ^22.11.0 || ^24.11.0",
   "projectFolderMinDepth": 1,
   "projectFolderMaxDepth": 1,
   "ensureConsistentVersions": true,

